lisa/features/gpu.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from enum import Enum
+from typing import Any, cast
+
+from lisa.feature import Feature
+from lisa.operating_system import Linux, Redhat, Ubuntu
+from lisa.tools import Uname, Wget
+from lisa.util import LisaException
+
+FEATURE_NAME_GPU = "Gpu"
+
+
+class compute_sdk(Enum):
+    GRID = 0
+    CUDA = 1
+
+
+class Gpu(Feature):
+    def __init__(self, node: Any, platform: Any) -> None:
+        super().__init__(node, platform)
+        self._log = self._node.log
+
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_GPU
+
+    def _install_grid_driver(self, version: str) -> None:
+        self._log.info("Starting GRID driver installation")
+
+    def _install_cuda_driver(self, version: str = "10.1.105-1") -> None:
+        self._log.info("Starting CUDA driver installation")
+        cuda_repo = ""
+        linux_os: Linux = cast(Linux, self._node.os)
+
+        # CUDA driver installation for redhat distros
+        if isinstance(self._node.os, Redhat):
+            cuda_repo_pkg = f"cuda-repo-rhel7-{version}.x86_64.rpm"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/"
+                f"compute/cuda/repos/rhel7/x86_64/{cuda_repo_pkg}"
+            )
+            linux_os = Redhat(self._node)
+
+        # CUDA driver installation for Ubuntu distros
+        elif isinstance(self._node.os, Ubuntu):
+            os_version = self._node.os.get_os_version()
+            cuda_repo_pkg = f"cuda-repo-ubuntu{os_version}//./_{version}_amd64.deb"
+            cuda_repo = (
+                "http://developer.download.nvidia.com/compute/"
+                f"cuda/repos/ubuntu{os_version}//.//x86_64/${cuda_repo_pkg}"
+            )
+            linux_os = Ubuntu(self._node)
+
+        else:
+            raise LisaException(
+                f"Distro {self._node.os._get_detect_string(self._node)}"
+                "not supported to install CUDA driver."
+            )
+
+        wget_tool = self._node.tools[Wget]
+        # download the cuda driver at /tmp/
+        wget_tool.get(cuda_repo, "/tmp/", cuda_repo_pkg)
+        # install the cuda driver rpm
+        install_result = linux_os.install_packages(
```

Comment:
the above wget commands should be in the install_packages method. So it can install an package from a web url.

Reply:
> the above wget commands should be in the install_packages method. So it can install an package from a web url.

Done